src/server/replica.cc

```diff
@@ -751,21 +751,34 @@ void Replica::StableSyncDflyFb(Context* cntx) {
 
   JournalReader reader{&ps, 0};
   JournalExecutor executor{&service_};
-  while (!cntx->IsCancelled()) {
-    auto res = reader.ReadEntry();
-    if (!res) {
-      cntx->ReportError(res.error(), "Journal format error");
-      return;
+
+  while (true) {
+    TranactionData tx_data;
+    bool cancelled = cntx->IsCancelled();
+    while (!cancelled) {
+      auto res = reader.ReadEntry();
+      if (!res) {
+        cntx->ReportError(res.error(), "Journal format error");
+        return;
+      }
+      tx_data.UpdateFromParsedEntry(std::move(*res));
+      if (tx_data.execute == true) {
+        break;
+      }
+      cancelled = cntx->IsCancelled();
+    }
+    if (cancelled) {
+      break;
     }
-    ExecuteEntry(&executor, std::move(res.value()));
+    ExecuteEntry(&executor, std::move(tx_data));
```

Comment:
Moving cancellation to a separate variable and then updating it makes it only more difficult.  `ExecuteEntry` should unblock either way in the future on cancellation, so we can (In theory) make both loops just while `!cntx->IsCancelled()`

Reply:
Ok I will change now ExecuteEntry to get the cntx and return if it is canceleed

Reply:
Its more difficult than that, we can leave it for later - it should unblock all the barriers inside the stable state error handler

Reply:
yes of course I know I need to unblock the barrier but this is for another PR
I did another PR in helio for barrier class with cancle function